client/src/components/GetDetails/index.js

```diff
@@ -129,7 +130,22 @@ class GetDetails extends Component {
         pattern: this.state.selected_patterns,
       };
 
-      if (cookie) {
+      if (inputs.typeId === '') {
+        axios.post('/add-type', { name: inputs.type.name, shortcut: 'New Type' }).then((res) => {
```

Comment:
oh I see you are using type here but still how u handle adding the type id in back end while you don't have this value in the request

Reply:
if you did not get what I mean, check this endpoint on BE, match FE with BE please